### ISSUE TITLE
IDE0072: False positive when using switch expression over a Nullable<T>, where both null and T cases are present

### ISSUE THREAD

**Issue by airbreather** [dotnet/roslyn#50983]
**Version Used**: 
.NET SDK 5.0.102

**Steps to Reproduce**:
`dotnet build` the following:
### Program.cs
```csharp
double? DivideByTwo(ulong? x)
{
    return x switch
    {
        null => null,
        ulong val => val / 2.0,
    };
}
```

### ConsoleApp0.csproj
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
  </PropertyGroup>

</Project>
```

### .editorconfig
```editorconfig
[*.cs]
dotnet_diagnostic.IDE0072.severity = warning
```
**Expected Behavior**:
IDE0072 is not raised: all cases are handled

**Actual Behavior**:
IDE0072 is raised.  The suggested fix is to add a default case (discard) at the end.

Based on https://github.com/dotnet/roslyn/issues/50982#issuecomment-772714446, I suspect that this may have the same root cause as both that issue and #48876, but I'm logging it separately for now just in case it does happen to be different enough.
